config/app/src/main/java/com/google/samples/quickstart/config/java/MainActivity.java

```diff
@@ -87,6 +90,27 @@ public void onClick(View v) {
         mFirebaseRemoteConfig.setDefaultsAsync(R.xml.remote_config_defaults);
         // [END set_default_values]
 
+        // [START add_config_update_listener]
+        mFirebaseRemoteConfig.addOnConfigUpdateListener(new ConfigUpdateListener() {
+            @Override
+            public void onUpdate(ConfigUpdate configUpdate) {
+                Log.d(TAG, "Updated keys: " + configUpdate.getUpdatedKeys());
+
+                mFirebaseRemoteConfig.activate().addOnCompleteListener(new OnCompleteListener<Boolean>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Boolean> task) {
+                        displayWelcomeMessage();
+                    }
+                });
+            }
+
+            @Override
+            public void onError(FirebaseRemoteConfigException error) {
+                Log.e(TAG, "Error: " + error.getCode() + ": " + error.getMessage(), error);
```

Comment:
```suggestion
                Log.w(TAG, "Config update error with code: " + error.getCode(), error);
```

Can be simplified since we're using the overloaded log method that takes the error.